### PR TITLE
[Bug](drop) fix drop function has error when it's not exists

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

```
drop function xxx
drop function if exists xxx
```
as when drop UDTF function, we need drop two function, one is normal, another is the _outer function.
getFunction() should call firstly and always.  as dropFunctionImpl() will remove it from env, then we can't find it anymore.
so can't check if (ifExists) firstly, when it's false, we have no change call getFunction()
ifExists should only used to check whether throw execption.

Issue Number: close #xxx

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 1]
```diff
@@ -748,14 +748,16 @@ public synchronized void replayAddFunction(Function function) {
...
-                udfFunction = getFunction(function);
-            } catch (AnalysisException e) {
-                // ignore it, as drop it if exist, so can't sure it must exist
+        try {
+            // here we must first getFunction, as dropFunctionImpl will remove it
+            udfFunction = getFunction(function);
+        } catch (AnalysisException e) {
+            if (!ifExists) {
+                throw new UserException(e);
             }
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:757]: !ifExists just return

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**